File: src/main/java/com/project/budgetapp/models/Expense.java
package com.project.budgetapp.models;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity(name="expenses")
public class Expense {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long expense_id;
    private String timestamp;
    private long category_id;
    private long account_id;
    private double amount;

    public Expense() {
    }

    public long getExpense_id() {
        return expense_id;
    }

    public void setExpense_id(long expense_id) {
        this.expense_id = expense_id;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(String timestamp) {
        this.timestamp = timestamp;
    }

    public long getCategory_id() {
        return category_id;
    }

    public void setCategory_id(long category_id) {
        this.category_id = category_id;
    }

    public long getAccount_id() {
        return account_id;
    }

    public void setAccount_id(long account_id) {
        this.account_id = account_id;
    }

    public double getAmount() {
        return amount;
    }

    public void setAmount(double amount) {
        this.amount = amount;
    }
}

File: Front-end/src/app/models/expense.ts
export class NewExpense {
    amount: number;
    category_id: number;
    name?: string;
    description?: string;
    account_id: number;

    constructor(amount: number, category_id: number, account_id: number, name?: string, description?: string) {
        this.amount = amount
        this.category_id = category_id;
        this.account_id = account_id;
        this.name = name;
        this.description = description;
    }
}


export class Expense extends NewExpense{
    expense_id: number;
    timestamp: string;
    category_id: number;
    account_id: number;
    amount: number;
    description: string;
    name: string;

    constructor(expense_id: number,
        timestamp: string,
        category_id: number,
        account_id: number,
        amount: number,
        name: string,
        description: string,
        category: string) {
            super(amount, category_id, account_id);

            this.expense_id = expense_id;
            this.expense_id = expense_id;
            this.name = name;
            this.timestamp = timestamp;
            this.category_id = category_id;
            this.account_id = account_id;
            this.amount = amount;
            this.description = description;
        }
}